modules/navier_stokes/src/postprocessors/PressureDrop.C

```diff
@@ -0,0 +1,339 @@
+    // and the delimitation would be within a refined element's face
+    if (status_known)
+      break;
+  }
+
+  if (upstream)
+  {
+    // Integration loops are different for FE and FV at this point
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_upstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_upstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_upstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_upstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+  // Downstream contributions
+  else
+  {
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_downstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_downstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_downstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_downstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+}
+
+Real
+PressureDrop::computeFaceInfoWeightedPressureIntegral(const FaceInfo * const fi) const
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  mooseAssert(fi, "We should have a face info in " + name());
+
+  const bool correct_skewness =
+      _weight_interp_method == Moose::FV::InterpMethod::SkewCorrectedAverage;
+  auto elem_arg = std::make_tuple(_current_elem, _qp, _qrule);
+  mooseAssert(_qp == 0, "Only one quadrature point");
+
+  if (_weighting_functor)
+  {
+    const auto ssf =
+        Moose::FaceArg({fi,
+                        Moose::FV::limiterType(_weight_interp_method),
+                        MetaPhysicL::raw_value((*_weighting_functor)(elem_arg)) * fi->normal() > 0,
+                        correct_skewness,
+                        nullptr});
+    const auto face_weighting = MetaPhysicL::raw_value((*_weighting_functor)(ssf));
+    return fi->normal() * face_weighting * _pressure(ssf);
+  }
+  else
+  {
+    // External faces for the pressure
+    if (!fi->neighborPtr() || !_pressure.hasBlocks(fi->neighborPtr()->subdomain_id()))
+    {
+      const auto ssf = Moose::FaceArg(
+          {fi, limiterType(_weight_interp_method), true, correct_skewness, _current_elem});
+      return _pressure(ssf);
+    }
+    else
+    {
+      const auto ssf = Moose::FaceArg(
+          {fi, Moose::FV::limiterType(_weight_interp_method), true, correct_skewness, nullptr});
+      return _pressure(ssf);
+    }
+  }
```

Comment:
```suggestion
  const auto ssf =
      Moose::FaceArg({fi,
                      Moose::FV::limiterType(_weight_interp_method),
                      MetaPhysicL::raw_value((*_weighting_functor)(elem_arg)) * fi->normal() > 0,
                      correct_skewness,
                      nullptr});
  if (_weighting_functor)
  {
    const auto face_weighting = MetaPhysicL::raw_value((*_weighting_functor)(ssf));
    return fi->normal() * face_weighting * _pressure(ssf);
  }
  else
    return _pressure(ssf);
```

Reply:
doesnt work for t he second case, cant use weighting functor to determine upstream-ness if not defined